solidity/contracts/PausableReentrancyGuard.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+// adapted from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+abstract contract PausableReentrancyGuardUpgradeable is Initializable {
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = 2;
+    uint256 private constant _PAUSED = 3;
+
+    uint256 private _status;
+
+    function __PausableReentrancyGuard_init() internal onlyInitializing {
+        _status = _NOT_ENTERED;
+    }
+
+    function _pause() internal {
+        _status = _PAUSED;
+    }
+
+    function _unpause() internal {
```

Comment:
slightly ridiculous but maybe put a `require(_status == _PAUSED)` here. Otherwise whoever has the authority to call `_unpause` will be able to reenter. Which in the Mailbox's case is the owner, so it's not a big deal, but probably best for completeness to have the check

Reply:
Same for pause actually-- we should require that the status isn't entered